gadk/elements.py

```diff
@@ -166,6 +168,49 @@ def step_extension(self, step: Dict) -> Dict:
         return step
 
 
+class CacheStep(UsesStep):
+    """A step that uses the cache action.
+
+    See https://github.com/marketplace/actions/cache
+    """
+
+    def __init__(
+        self,
+        name: str,
+        paths: Iterable[Union[Path, str]],
+        key: str,
+        restore_keys: Optional[Iterable[str]] = None,
+        version: str = "v3",
+        **kwargs,
+    ) -> None:
+        with_args = {"path": "/n".join(str(path) for path in paths), "key": key}
+        if restore_keys is not None:
+            with_args["restore-keys"] = "\n".join(restore_keys)
+
+        super().__init__(
+            name=name, action=f"actions/cache@{version}", with_args=with_args, **kwargs
+        )
+
+    @classmethod
+    def simple(
+        cls,
+        name: str,
+        path: Union[Path, str],
+        slug: str,
+        hash_files: Iterable[Union[str, Path]],
+        **kwargs,
+    ):
+        # Assume that filenames don't contain quotes
+        hash_files_quoted = ", ".join(f"'{filename}'" for filename in hash_files)
+        return cls(
+            name,
+            [path],
+            f"{slug}-${{{{ hashFiles({hash_files_quoted}) }}}}",
+            [f"{slug}-"],
+            **kwargs,
+        )
```

Comment:
It's not a different kind of step, it's just a utility to handle the most common use case. I didn't want to make this the constructor because there are many other valid ways to use `CacheStep`, but our cases all follow this pattern, which is a common one.